src/connect/connect.js

```diff
@@ -29,16 +30,24 @@ function match(arg, factories) {
   return undefined
 }
 
+function strictEqual(a, b) { return a === b }
+
 export function buildConnectOptions(
   mapStateToProps,
   mapDispatchToProps,
   mergeProps,
   {
-    mapStateToPropsFactories = defaultMapStateToPropsFactories,
-    mapDispatchToPropsFactories = defaultMapDispatchToPropsFactories,
-    mergePropsFactories = defaultMergePropsFactories,
-    selectorFactory = defaultSelectorFactory,
     pure = true,
+    areStatesEqual = strictEqual,
+    areOwnPropsEqual = shallowEqual,
+    areStatePropsEqual = shallowEqual,
+    areMergedPropsEqual = shallowEqual,
+    factories: {
```

Comment:
As I'm writing the docs for the new connect features, i'm realizing the 4 factories are different than the other options; they don't make as much sense to use at the individual call sites like the other options. They're intention is to be used in a custom wrapper function like 'customConnect`. That's why I'm trying to separate them from the other options. Maybe they shouldn't be in options at all. Maybe I should use a customization pattern like [redux-thunk](https://github.com/gaearon/redux-thunk/blob/master/src/index.js) or [reselect's createSelector](https://github.com/reactjs/reselect/blob/master/src/index.js#L66). Any thoughts on that?